sklearn/preprocessing/data.py

```diff
@@ -2472,3 +2475,239 @@ def quantile_transform(X, axis=0, n_quantiles=1000,
     else:
         raise ValueError("axis should be either equal to 0 or 1. Got"
                          " axis={}".format(axis))
+
+
+class UnaryEncoder(BaseEstimator, TransformerMixin):
+    """Encode ordinal integer features using a unary scheme.
+
+    The input to this transformer should be a matrix of integers, denoting
+    the values taken on by ordinal (discrete) features. The output will be
+    a sparse matrix where each column corresponds to one possible value of one
+    feature. It is assumed that input features take on values in the range
+    [0, n_values).
```

Comment:
While this interval notation is well-known, I think "from 0 to n_values - 1" is much clearer.

Reply:
done